app/api/auth/product-page/page.tsx

```diff
@@ -0,0 +1,50 @@
+'use client';
+
+import React, { Component } from 'react';
+
+class AddProduct extends Component {
+  state = {
+    productName: '',
+    showText: false
```

Comment:
Since we are using Typescript, you can give types to your state. Here are some examples

https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/hooks/
https://www.robinwieruch.de/typescript-react-usestate/